shared/fs/banner/conflict-banner.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Flow from '../../util/flow'
+import * as Types from '../../constants/types/fs'
+
+/*
+ * This banner is used as part of a List2 in fs/row/rows.js, so it's important
+ * to keep height stable, thus all the height/minHeight/maxHeight in styles.
+ * Please make sure the height is still calculated in getHeight when layout
+ * changes.
+ *
+ */
+export const getHeight = (conflictState: Types.ConflictState): number => (conflictState === 'none' ? 0 : 50)
+
+type Props = {
+  conflictState: Types.ConflictState,
+  onStartResolving: () => void,
+  onSeeOtherView: () => void,
+  isUnmergedView: boolean,
+  onFinishResolving: () => void,
+  onFeedback: () => void,
+  onHelp: () => void,
+}
+
+const getMessage = (conflictState, isUnmerged): string => {
+  switch (conflictState) {
+    case 'in-conflict-stuck':
+      return (
+        'Your changes to this folder conflict with changes made to this' +
+        ' folder on the server. Automatic conflict resolution has failed, so' +
+        ' you may need to manually resolve the conflict. This is not' +
+        ' supposed to happen!'
+      )
+    case 'in-conflict-not-stuck':
+      return (
+        'Your changes to this folder conflict with changes made to this' +
+        ' folder on the server. We are trying to fix it automatically, but' +
+        ' you may need to manually resolve the conflict. This is not' +
+        ' supposed to happen!'
+      )
+    case 'in-manual-resolution':
+      return isUnmerged
+        ? 'This is your local view of the conflicted folder.'
+        : "This is the server's view of the conflicted folder."
```

Comment:
Several thoughts on the wording:

1. I wonder if we should avoid mentioning "server", not only because of #design's opinion, but also that it'd probably give a wrong impression that server can see the content somehow. How about something like "other devices's view".

2. I feel like `in-conflict-stuck` should say "you need to manually resolve" instead of "you may need to manually resolve". `in-conflict-not-stuck` should probably not mention manual resolution. I think neither should say "This is not supposed to happen" as that sounds like a bug, but I can be convinced otherwise.

3. It seems `in-manual-resolution` could use some more text -- something like suggesting user to click "finish resolving" when they're done when in server view, and "go to server view to resolve it" when in local view.

Reply:
Also, since conflict is in TLF level, perhaps it'd be helpful to say "/keybase/private/jakob223" instead of "this folder". Related, we could probably make it more clear that user is currently in a subfolder of the conflict TLF, so they don't mistakenly mark a resolution as finished when they are only done with a particular dir within the TLF.

Reply:
> I think neither should say "This is not supposed to happen" as that sounds like a bug, but I can be convinced otherwise
This isn't supposed to happen, though - and we want to know about it when it happens. If we don't say something like that, I feel like users may not have any reason to click "let us know".

Reply:
> 

Also, since conflict is in TLF level, perhaps it'd be helpful to say "/keybase/private/jakob223" instead of "this folder". Related, we could probably make it more clear that user is currently in a subfolder of the conflict TLF, so they don't mistakenly mark a resolution as finished when they are only done with a particular dir within the TLF.

Hmm, this is a good point. I wonder if we should only let them click finish from the TLF itself.. WDYT of that?

Reply:
> `in-conflict-not-stuck` should probably not mention manual resolution

I think it should, because I imagine that most CR conflicts will continue to be unresolved after 10 tries if they are unresolved after one try. Maybe we should lower the limit from 10 to 5 or something too..

Reply:
> This isn't supposed to happen, though - and we want to know about it when it happens. If we don't say something like that, I feel like users may not have any reason to click "let us know".

The "not stuck" situation isn't really not supposed to happen right? It happens all the time; but just get resolved automatically quickly.

> Hmm, this is a good point. I wonder if we should only let them click finish from the TLF itself.. WDYT of that?

That sounds good with me. I think we should be careful to not mention the term TLF though -- most users don't know what it is and we haven't introduced that concept in GUI before.

> I think it should, because I imagine that most CR conflicts will continue to be unresolved after 10 tries if they are unresolved after one try. Maybe we should lower the limit from 10 to 5 or something too..

Related to the first point; I was under the impression that this happens underneath automatically all the time. For example when you're behind the server and have the local chages in the journal, and they get "rebased" to latest version before being pushed. So it seems to me most of the "not stuck" conflict will never need manual intervention. I could be wrong though; maybe @strib could clarify

Reply:
As I understand it, the CR state would not change to `in-conflict-not-stuck` in that automatic "rebase" situation. Only once the CR fails once and we record a panic or an error, would we enter the conflict state. At that point CR might try again (when another change happens in the TLF, or when KBFS restarts after having died due to a panic) and maybe succeed if it failed due to running out of memory, though likely fail for the same reason as before. Once this trying again has happened 10 times, we enter `in-conflict-stuck` and auto-fail CR.

Reply:
@songgao is right, it switches to `in-conflict-not-stuck` as soon as it notices a conflict, and before it gets stuck.  But, there might not be a notification to the UX in that case until it becomes stuck, so you'd only see it if you happen to poll the favorites during that time.  I don't recommend having a badge or a banner in that state right now, as it doesn't require any user interaction and should be a very brief state, unless it becomes stuck.